ideal_device/io/tlb_manager.h

```diff
@@ -0,0 +1,183 @@
+/*
+ * SPDX-FileCopyrightText: (c) 2023 Tenstorrent Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#pragma once
+
+/*
+
+Questions:
+
+- Do we need per arch TLB manager (for example for UC/WC distribution)?
+
+- Do we need to have index in TLBWindow in order to return it properly?
+
+*/
+
+#include "pci_device.h"
+#include "tlb_window.h"
+
+namespace tt::umd {
+
+class TLBManager {
+
+public:
+    virtual TLBWindow get_tlb_window(tt_xy_pair core) = 0;
+    
+    virtual TLBWindow get_wc_tlb_window() = 0;
+
+    virtual TLBWindow get_uc_tlb_window() = 0;
```

Comment:
uc vs wc can be a parameter of the same function

Reply:
agree